src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```diff
@@ -0,0 +1,203 @@
+import { Component, ElementRef, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { AttachmentService } from 'app/lecture/attachment.service';
+import * as PDFJS from 'pdfjs-dist';
+import 'pdfjs-dist/build/pdf.worker';
+import { Attachment } from 'app/entities/attachment.model';
+import { AttachmentUnit } from 'app/entities/lecture-unit/attachmentUnit.model';
+import { AttachmentUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/attachmentUnit.service';
+import { onError } from 'app/shared/util/global.utils';
+import { AlertService } from 'app/core/util/alert.service';
+
+@Component({
+    selector: 'jhi-pdf-preview-component',
+    templateUrl: './pdf-preview.component.html',
+    styleUrls: ['./pdf-preview.component.scss'],
+})
+export class PdfPreviewComponent implements OnInit, OnDestroy {
+    @ViewChild('pdfContainer', { static: true }) pdfContainer: ElementRef;
+    @ViewChild('enlargedCanvas') enlargedCanvas: ElementRef;
+    attachment?: Attachment;
+    attachmentUnit?: AttachmentUnit;
+    isEnlargedView: boolean = false;
+    currentPage: number = 1;
+    totalPages: number = 0;
+
+    constructor(
+        private route: ActivatedRoute,
+        private attachmentService: AttachmentService,
+        private attachmentUnitService: AttachmentUnitService,
+        private alertService: AlertService,
+    ) {}
+
+    ngOnInit() {
+        this.route.data.subscribe((data) => {
+            if ('attachment' in data) {
+                this.attachment = data.attachment;
+                if (this.attachment?.id) {
+                    this.attachmentService.getAttachmentFile(this.attachment.id).subscribe({
+                        next: (blob: Blob) => this.loadPdf(URL.createObjectURL(blob)),
+                        error: (error) => onError(this.alertService, error),
```

Comment:
I think that a better approach would be to create a `NotFoundAlertException` on the server side (have a look at the `BadRequestAlertException`) and directly pass the translation key from the server side. 

Explicitly catching the error here should not be required then.

Reply:
Attachment attachment = attachmentRepository.findById(attachmentId)
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Attachment not found with id: " + attachmentId));


I already have a check like this actually in FileResource, do you mean something like this?

Reply:
Yes, we could directly provide the translation key here aswell (as done in the `BadRequestAlertException`). The class for a `NotFoundAlertException` does not exist yet.

Reply:
The error handling is quite inconsistent in Artemis, as far as I am aware we wanted to move in the direction to use more `...AlertExceptions` as we can easily modify the error handling in a few classes then if needed (which might be the case since the client side `alert.service.ts` might also benefit from a refactoring)